src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -5736,13 +5897,12 @@ static MyCollection<object> F2(int x, object y)
             verifier.VerifyIL("Program.F0",
                 """
                 {
-                  // Code size       17 (0x11)
+                  // Code size       16 (0x10)
                   .maxstack  1
-                  IL_0000:  ldc.i4.0
-                  IL_0001:  newarr     "string"
-                  IL_0006:  newobj     "System.ReadOnlySpan<string>..ctor(string[])"
-                  IL_000b:  call       "MyCollection<string> MyCollectionBuilder.Create<string>(System.ReadOnlySpan<string>)"
-                  IL_0010:  ret
+                  IL_0000:  call       "string[] System.Array.Empty<string>()"
+                  IL_0005:  newobj     "System.ReadOnlySpan<string>..ctor(string[])"
+                  IL_000a:  call       "MyCollection<string> MyCollectionBuilder.Create<string>(System.ReadOnlySpan<string>)"
+                  IL_000f:  ret
                 }
                 """);
             verifier.VerifyIL("Program.F1",
```

Comment:
add test where the destination is an `int*[][]`  this should be able to use `Array.Empty<int*[]>` (inner type is pointer, outer is array).